feat: make `IsOptional` conditional

yitzchak-ben-ezra-ecoplant opened the pull request:
## Description
<!-- A clear and concise description what these changes does. -->
**There is already an open approved PR which is not merged!
This is a duplicate just to get this merged already**
https://github.com/typestack/class-validator/pull/196
AND
https://github.com/typestack/class-validator/issues/651

## Checklist
<!-- Replace  the [ ] with [x] to check the boxes. -->
- [x] the pull request title describes what this PR does (not a vague title like `Update index.md`)
- [x] the pull request targets the *default* branch of the repository (`develop`)
- [x] the code follows the established code style of the repository
  -[x] `npm run prettier:check` passes
  -[x] `npm run lint:check` passes
- [x] tests are added for the changes I made (if any source code was modified)
- [x] documentation added or updated
- [x] I have run the project locally and verified that there are no errors

### Fixes
<!-- If there is no issue being resolved, open one before creating this pull request. -->
<!-- If the PR doesn't fully resolve the issue, replace 'fixes' with 'references'. -->
fixes #[issue number], fixes #[issue number]

yitzchak-ben-ezra-ecoplant committed:
feat: make `IsOptional` conditional

yitzchak-ben-ezra-ecoplant committed:
feat: make `IsOptional` conditional

erabl-identos commented on src/decorator/common/IsOptional.ts:
```diff
@@ -4,22 +4,37 @@ import { ValidationTypes } from '../../validation/ValidationTypes';
       type: ValidationTypes.CONDITIONAL_VALIDATION,
       target: object.constructor,
-      propertyName: propertyName,
+      propertyName,
```
```suggestion
      propertyName: propertyName,
```

yitzchak-ben-ezra-ecoplant commented:
@erabl-identos please complete the review so it gets merged...

erabl-identos commented on test/functional/conditional-validation.spec.ts:
```diff
@@ -92,4 +92,72 @@ describe('conditional validation', () => {
+    });
+  });
+
+  it('should validate a property when condition returns true and value is not supplied', function () {
```
when the condition returns true (meaning the property is optional), the validation should _not_ happen if the property was not supplied.

erabl-identos reviewed (commented):
looking forward to use this feature

yitzchak-ben-ezra-ecoplant committed:
fix: spec description

yitzchak-ben-ezra-ecoplant committed:
fix: spec description

yitzchak-ben-ezra-ecoplant commented:
@erabl-identos ready for approval

erabl-identos commented:
According to Github at least one approval must be provided by somebody with write access.

Maybe @NoNameProvided can move this forward?

yitzchak-ben-ezra-ecoplant commented:
@NoNameProvided Please close this.. we really need this.. and it's waiting for more than 2 years...

yitzchak-ben-ezra-ecoplant commented:
Maybe @pleerock / @vlapo - Do you have write access and can review this minor PR?

ybeapps commented:
Looks like this repo is dead. No maintainers comments in this minor PR since it’s been created.. almost 2 weeks..
Only the bot doing any work here..
Too bad.. it’s a good repo, let others who have more time to maintain it for the world..

yitzchak-ben-ezra-ecoplant committed:
Update package.json

yitzchak-ben-ezra-ecoplant committed:
Update package.json

Update dependabot.yml

dummy change to enable dependabot

yitzchak-ben-ezra-ecoplant committed:
Update package.json

yitzchak-ben-ezra-ecoplant committed:
Create dependabot2.yml

yitzchak-ben-ezra-ecoplant committed:
Delete dependabot2.yml

yitzchak-ben-ezra-ecoplant committed:
Update dependabot.yml

dummy change to enable dependabot

dependabot[bot] committed:
build(deps-dev): bump typescript from 4.2.2 to 4.2.4

Bumps [typescript](https://github.com/Microsoft/TypeScript) from 4.2.2 to 4.2.4.
- [Release notes](https://github.com/Microsoft/TypeScript/releases)
- [Commits](https://github.com/Microsoft/TypeScript/compare/v4.2.2...v4.2.4)

Signed-off-by: dependabot[bot] <support@github.com>

yitzchak-ben-ezra-ecoplant committed:
Merge pull request #2 from yitzchak-ben-ezra-ecoplant/dependabot/npm_and_yarn/typescript-4.2.4

build(deps-dev): bump typescript from 4.2.2 to 4.2.4

yitzchak-ben-ezra-ecoplant committed:
Update auto-approve-dependabot-workflow.yml

yitzchak-ben-ezra-ecoplant committed:
build: bump version to 0.13.2

yitzchak-ben-ezra-ecoplant commented:
Closing this PR since this repo is DEAD!!
I've forked this repo.. and will open it for public edits.. whoever wants to help maintain it - your welcome!
https://github.com/yitzchak-ben-ezra-ecoplant/class-validator-forked

NoNameProvided commented:
https://github.com/typestack/class-validator/pull/932
